pkg/kapi/kapi.go

```diff
@@ -223,25 +222,33 @@ func ScaleDeployment(kcontext, namespace, deploymentName string, replicas int) e
 	err = wait.PollImmediate(kconst.APICallRetryInterval, ReasonableMutateTime, func() (bool, error) {
 		scale, err := client.AppsV1().Deployments(namespace).GetScale(context.Background(), deploymentName, meta.GetOptions{})
 		if err != nil {
-			klog.Warningf("failed getting deployment scale, will retry: %v", err)
-			return false, nil
+			if IsRetryableAPIError(err) {
+				klog.Warningf("failed getting %q deployment scale, will retry: %v", deploymentName, err)
+				return false, nil
+			}
+			klog.Info("non-retryable failure while getting %q deployment scale: %v", deploymentName, err)
```

Comment:
```suggestion
			klog.Infof("non-retryable failure while getting %q deployment scale: %v", deploymentName, err)
```
Should this be a `Warningf` instead of an `Infof`?

Reply:
like below -

i think we could actually return the error here and effectively stop, otherwise, we'd continue retrying - something like:

> return false, fmt.Errorf("non-retryable failure while getting %q deployment scale: %v", deploymentName, err)